pydantic_extra_types/coordinate.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+from typing import Any, ClassVar, Tuple, Union
+
+from pydantic import GetCoreSchemaHandler, GetJsonSchemaHandler, dataclasses
+from pydantic._internal import _repr
+from pydantic.json_schema import JsonSchemaValue
+from pydantic_core import ArgsKwargs, PydanticCustomError, core_schema
+
+CoordinateValueType = Union[str, int, float]
+
+
+class Latitude(float):
+    min: ClassVar[float] = -90.00
+    max: ClassVar[float] = 90.00
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.float_schema(ge=cls.min, le=cls.max)
+
+
+class Longitude(float):
+    min: ClassVar[float] = -180.00
+    max: ClassVar[float] = 180.00
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.float_schema(ge=cls.min, le=cls.max)
+
+
+@dataclasses.dataclass
+class Coordinate(_repr.Representation):
+    _NULL_ISLAND: ClassVar[tuple[float, float]] = 0.0, 0.0
+
+    latitude: Latitude
+    longitude: Longitude
+
+    @classmethod
+    def __get_pydantic_json_schema__(
+        cls, core_schema: core_schema.CoreSchema, handler: GetJsonSchemaHandler
+    ) -> JsonSchemaValue:
+        field_schema: dict[str, Any] = handler(core_schema)
+        field_schema.update(format='coordinate')
+        return field_schema
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.no_info_wrap_validator_function(
+            cls._parse_args,
+            core_schema.no_info_wrap_validator_function(
+                cls._parse_str,
+                core_schema.chain_schema(
+                    [
+                        core_schema.no_info_wrap_validator_function(
+                            cls._parse_tuple, handler.generate_schema(Tuple[float, float])
+                        ),
+                        handler(source),
+                    ]
+                ),
+            ),
+        )
+
+    @classmethod
+    def _parse_args(cls, value: Any, handler) -> Any:
+        if isinstance(value, ArgsKwargs) and not value.kwargs:
+            n_args = len(value.args)
+            if n_args == 0:
+                value = cls._NULL_ISLAND
+            elif n_args == 1:
+                value = value.args[0]
```

Comment:
On my branch I've modified this part to add a default latitude, otherwise we'll get an error

Reply:
This is an error about the error case that is needed. We get an error for case like `(20.0,)` that is a tuple with a single value. This case is in tests as invalid.

Reply:
You’re right but maybe we can raise an error directly, WDYT?